csrc/polymorphic_value.h

```diff
@@ -373,6 +373,21 @@ inline PolymorphicValue toTensor(
       false, "PolymorphicValue toTensor not implemented for ", x.type().name());
 }
 
+// Convert PolymorphicValue to c10::Scalar
+inline c10::Scalar toScalar(const PolymorphicValue& x) {
```

Comment:
Can this be just
```C++
if (x.is<std::complex<double>>()) {
  return (c10::complex<double>)x;
} else {
  return x;
}
```

Reply:
That gives a conversion error from `PolymorphicValue` to `c10::Scalar`

Reply:
Does this work?
```C++
if (x.is<std::complex<double>>()) {
  return (c10::complex<double>)x;
} else {
  return (c10::Scalar)x;
}
```

Reply:
Yes, that works. Although in that case, we will not be checking if the input can actually be converted to a Scalar (i.e, if x is one of `at::Tensor, <std::vector>, StructHandle, Pointer, Opaque`) so I prefer the explicit checks. We can keep this implementation and be careful about the usage as well.

Additionally, can we make this more general? It would be useful in the `TernaryOp where` when the condition input is a tensor instead of a scalar. Right now, the `evaluate` method does not work for tensors: Example test: https://github.com/NVIDIA/Fuser/blob/e30b6bbc353f1595c8b5c3337626b8a90c80b16b/test/test_gpu_view.cpp#L2177
In these cases, `at::where` expects the second and third argument to either be a tensor or scalar.
```
inline std::variant<at::Tensor, c10::Scalar> toTensorOrScalar(const PolymorphicValue& x)
```

Reply:
Sounds good, let's do it!

Reply:
Sticking to `toScalar` since `at::where` will require explicit conversion to either a tensor/scalar so the evalaute method will be just as verbose. Will modify `toScalar` if needed in a separate PR.